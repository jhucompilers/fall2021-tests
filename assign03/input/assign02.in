PROGRAM assign02;
    TYPE P = RECORD x: INTEGER; END;

    TYPE R = RECORD x, y: INTEGER; END;
    TYPE Q = RECORD a, b: P; END;

    VAR a: ARRAY 5 OF R;
    VAR b: ARRAY 5 OF Q;
BEGIN
    -- a := b;
    a[0].x := 4;
    b[0].a.x := 5;
END.
