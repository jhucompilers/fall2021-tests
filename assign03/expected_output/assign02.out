1,VAR,x,INTEGER
0,TYPE,P,RECORD (INTEGER)
1,VAR,x,INTEGER
1,VAR,y,INTEGER
0,TYPE,R,RECORD (INTEGER x INTEGER)
1,VAR,a,RECORD (INTEGER)
1,VAR,b,RECORD (INTEGER)
0,TYPE,Q,RECORD (RECORD (INTEGER) x RECORD (INTEGER))
0,VAR,a,ARRAY 5 OF RECORD (INTEGER x INTEGER)
0,VAR,b,ARRAY 5 OF RECORD (RECORD (INTEGER) x RECORD (INTEGER))
